### PR TITLE
Add rule S109 (based on 'no-magic-numbers') for TS/JS (new rule for JS)

### LINKED ISSUE #1341: Add rule S109 (based on 'no-magic-numbers') for TS/JS (new rule for JS)

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes #1341

### CODE REVIEW

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<p>A magic number is a number that comes out of nowhere, and is directly used in a statement. Magic numbers are often used, for instance to limit the
+number of iterations of a loops, to test the value of a property, etc.</p>
```
Comment (yassin-kammoun-sonarsource) [inline javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html]: `a loops`

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+<p>Using magic numbers may seem obvious and straightforward when you're writing a piece of code, but they are much less obvious and straightforward at
+debugging time.</p>
+<p>That is why magic numbers must be demystified by first being assigned to clearly named variables before being used.</p>
+<p>-1, 0 and 1 are not considered magic numbers.</p>
```
Comment (yassin-kammoun-sonarsource) [inline javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html]: Just saying, I've seen other specs using a dedicated `Exceptions` section.

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html (hunk 1 above)
Comment (vilchik-elena) [reply javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html]: fixed in RSPEC, will be updated before release

File: javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html (hunk 2 above)
Comment (vilchik-elena) [reply javascript-checks/src/main/resources/org/sonar/l10n/javascript/rules/javascript/S109.html]: indeed, but I think it's ok like that here

File: javascript-checks/src/main/java/org/sonar/javascript/checks/NoMagicNumbersCheck.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.sonar.javascript.checks.annotations.TypeScriptRule;
+
+@TypeScriptRule
+@JavaScriptRule
+@Rule(key = "S109")
+public class NoMagicNumbersCheck extends EslintBasedCheck {
+
+  @Override
+  public List<Object> configurations() {
+    return Collections.unmodifiableList(Lists.newArrayList(new Config()));
```
[the remark this answers is not in the document]
Comment (vilchik-elena) [reply javascript-checks/src/main/java/org/sonar/javascript/checks/NoMagicNumbersCheck.java]: I will fix it in another PR

**Pull request merged**